### PR TITLE
Remove python 2.7

### PR DESCRIPTION
**Pull request by cunla**
- format based on flake8.
- fix GitHub workflow.
- remove python 2.7.

### CODE REVIEW

File: jsonpointer.py [hunk 1]
```diff
@@ -309,12 +295,13 @@ def join(self, suffix):
...
-        except:
+        except:  # noqa E722
             raise JsonPointerException("Invalid suffix")
 
-    def __truediv__(self, suffix): # Python 3
+    def __truediv__(self, suffix):  # Python 3
         return self.join(suffix)
-    __div__ = __truediv__ # Python 2
+
+    __div__ = __truediv__  # Python 2
```
Comment (stefankoegl) [inline jsonpointer.py:304]: Is that still needed when we no longer support Python 2.x?

Comment (danielm-cfa) [reply jsonpointer.py:304]: Good point, I'll remove

**Comment by cunla** [comment]
https://github.com/cunla/python-json-pointer/actions/runs/9275553066

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,4 @@ dist
 *.swp
 doc/_build
 *.egg-info
+.idea
```
Comment (stefankoegl) [inline .gitignore:9]: What's that? Never heard of it

File: jsonpointer.py [hunk 3]
```diff
@@ -32,38 +32,22 @@
 
 """ Identify specific nodes in a JSON document (RFC 6901) """
 
-from __future__ import unicode_literals
-
 # Will be parsed by setup.py to determine package metadata
 __author__ = 'Stefan Kögl <stefan@skoegl.net>'
-__version__ = '2.4'
+__version__ = '3.0.0'
```
Comment (stefankoegl) [inline jsonpointer.py:37]: Any specific reason to go for 3 digits? Is there some standard to follow?

File: .gitignore (hunk 2 above)
Comment (cunla) [reply .gitignore:9]: jetbrains IDEs create this folder.
This just prevents from adding it accidently.

File: jsonpointer.py (hunk 3 above)
Comment (cunla) [reply jsonpointer.py:37]: [semver](https://semver.org/)

**Pull request merged**